graph/graph.go

```diff
@@ -191,21 +188,24 @@ func (graph *Graph) Register(img *image.Image, layerData archive.ArchiveReader)
 	graph.driver.Remove(img.ID)
 
 	tmp, err := graph.mktemp("")
-	defer os.RemoveAll(tmp)
 	if err != nil {
 		return fmt.Errorf("mktemp failed: %s", err)
 	}
+	defer os.RemoveAll(tmp)
 
 	// Create root filesystem in the driver
 	if err := graph.driver.Create(img.ID, img.Parent); err != nil {
+		cleanup()
```

Comment:
@jfrazelle We thought this was the issue, but could not reproduce it: http://play.golang.org/p/1aHCqY9FZc works correctly. Not sure if we should actually merge this...